### PR TITLE
fix double entry issue

### PR DESCRIPTION
**Pull request by ICraig-CH**
- fix double entry issue by adding if statement to diverge the paths at saveObligedEntityType in psc_discrepancy.js- fixed issue with session data being retained when yoy go back to the start page
- fixed misc spacing issues that lint brought up

Jira ticket : https://companieshouse.atlassian.net/browse/FAML-700

### CODE REVIEW

File: test/server/routes/report.test.js [hunk 1]
```diff
@@ -508,6 +507,7 @@ describe('routes/report', () => {
       .set('Cookie', cookie)
       .send(clientPayload)
       .then(response => {
+        console.log(response);
```
Comment (CH-JonathanBlake) [inline test/server/routes/report.test.js:510]: Do you still want these?

Comment (ICraig-CH) [reply test/server/routes/report.test.js]: no, sorry I will remove them :)

File: server/services/psc_discrepancy.js [hunk 2]
```diff
@@ -38,19 +40,39 @@ class PscDiscrepancy {
...
-      },
-      url: `${this.server.baseUrl}/psc-discrepancy-reports`
-    });
-    logger.info('Service request to save obliged entity type, with payload: ', options);
-
-    const client = APIClientHelper.getAPIClient(oauthToken);
-    return client.pscDiscrepancyReport.createNewReport(obligedEntityTypeId).then(result => this.handleResponse(result));
+  saveObligedEntityType (oauthToken, req) {
+    const pscCache = cacheService.getCachedDataFromSession(req.session);
+    if (pscCache.passedDiscrepancyDetails === true) {
```
Comment (CH-JonathanBlake) [inline server/services/psc_discrepancy.js:45]: This feels like controller logic not service logic.
Is this still reusable in other places?
Renaming methods could be helpful, such as changing this to saveCachedReport. And moving the else into a createNewReport?

Comment (ICraig-CH) [reply server/services/psc_discrepancy.js:45]: I did it here because of the .then that occurs after the save is run

Comment (CH-JonathanBlake) [reply server/services/psc_discrepancy.js:45]: Fine create two methods then. my point is that I don't think it should be the responsibility of the service to decide if its in the Change your XXXX part of the journey

Comment (ICraig-CH) [reply server/services/psc_discrepancy.js:45]: I can try and split them up then :)

File: server/routes/report.js [hunk 3]
```diff
@@ -57,7 +57,12 @@ router.post('/report-a-discrepancy/obliged-entity/type', (req, res, next) => {
   logger.info('POST request to save obliged entity contact type, with payload: ', req.body);
   validator.isValidObligedEntityType(req.body, Object.keys(obligedEntityTypes))
     .then(_ => {
-      return pscDiscrepancyService.saveObligedEntityType(Utility.getOAuthToken(req), req);
+      const pscCache = cacheService.getCachedDataFromSession(req.session);
+      if (typeof pscCache.passedDiscrepancyDetails === 'undefined' || pscCache.passedDiscrepancyDetails === false) {
+        return pscDiscrepancyService.saveObligedEntityType(Utility.getOAuthToken(req), req);
+      } else {
+        return pscDiscrepancyService.updateObligedEntityType(Utility.getOAuthToken(req), req);
```
Comment (CH-JonathanBlake) [inline server/routes/report.js:64]: we already have the chache here, we could pass the self link in rather than the request which would be more similar with the signature we're already using for getReport.

File: server/routes/report.js [hunk 4]
```diff
@@ -53,11 +57,18 @@ router.post('/report-a-discrepancy/obliged-entity/type', (req, res, next) => {
...
+      }
     }).then(r => {
       const pscCache = cacheService.getCachedDataFromSession(req.session);
-      pscCache.appData.initialServiceResponse = r;
-      cacheService.setPscCache(req.session, pscCache);
+      if (typeof pscCache.passedDiscrepancyDetails === 'undefined' || pscCache.passedDiscrepancyDetails === false) {
```
Comment (CH-JonathanBlake) [inline server/routes/report.js:68]: This if is identical to the one in the previous when, we could combine these and remove this then.

Comment (ICraig-CH) [reply server/routes/report.js:68]: I need to wait for the save or update to be performed before doing this bit

File: server/services/psc_discrepancy.js [hunk 5]
```diff
@@ -38,19 +40,39 @@ class PscDiscrepancy {
     return client.pscDiscrepancyReport.getReportBySelfLink(selfLink).then(result => this.handleResponse(result));
   }
 
-  saveObligedEntityType (oauthToken, obligedEntityTypeId) {
+  saveObligedEntityType (oauthToken, req) {
     const options = Object.assign(this._getBaseOptions(), {
       method: 'POST',
       data: {
-        obliged_entity_type: obligedEntityTypeId,
+        obliged_entity_type: req.body.obligedEntityType,
```
Comment (CH-JonathanBlake) [inline server/services/psc_discrepancy.js:47]: could we revert this change, I think req was a hold over from when you needed to use the cache.

File: server/routes/report.js (hunk 4 above)
Comment (CH-JonathanBlake) [reply server/routes/report.js:68]: oh this is flattening the potential nested promises? 👍

Comment (ICraig-CH) [reply server/routes/report.js:68]: I think that is how it was designed but it causes errors if this is run after the update rather than the save so I had to put in that condition

File: server/services/psc_discrepancy.js (hunk 5 above)
Comment (ICraig-CH) [reply server/services/psc_discrepancy.js:47]: I will change it so it passes in the oauth token, self link and the new oe type 👍

**Review by ch-rfishwick — APPROVED** [review]

**Review by CH-JonathanBlake — APPROVED** [review]
